pkg/api/authz.go

```diff
@@ -268,24 +259,24 @@ func BaseAuthzHandler(ctlr *Controller) mux.MiddlewareFunc {
 				return
 			}
 
-			acCtrlr := NewAccessController(ctlr.Config)
+			aCtlr := NewAccessController(ctlr.Config)
 
 			var identity string
 
 			// anonymous context
-			acCtx := &localCtx.AccessControlContext{}
+			userAc := reqCtx.NewUserAccessControl()
 
 			// get username from context made in authn.go
 			if ctlr.Config.IsBasicAuthnEnabled() {
 				// get access control context made in authn.go if authn is enabled
-				acCtx, err = localCtx.GetAccessControlContext(request.Context())
+				userAc, err = reqCtx.UserAcFromContext(request.Context())
 				if err != nil { // should never happen
 					authFail(response, request, ctlr.Config.HTTP.Realm, ctlr.Config.HTTP.Auth.FailDelay)
 
 					return
 				}
 
-				identity = acCtx.Username
+				identity = userAc.GetUsername()
 			}
 
 			if request.TLS != nil {
```

Comment:
Is there any reason for this code to be in authz.go instead of authn.go? Authz should read already existing user information from userAc.

Reply:
yes, this is an edge case where authn is basically skipped, and user is authenticated using tls, by gorilrla package.

Reply:
no :(

Reply:
Why not?

Reply:
because in this case there is no authn.go.

this is the case where we have authz and tls mutual authentication and no authn(ldap/htpasswd/openid enabled.
auth is made by gorrilla, and in authz we take the identity from verified chains from the request.

our authn.go is not USEd as a middleware at all.

Reply:
It's still not clear to me why we can't do add and additional check to see if certificate authentication is enabled somewhere in https://github.com/project-zot/zot/blob/main/pkg/api/authn.go#L230, and if it is, and use this logic there.

And if certificate authentication is enabled apply the authentication handler on the router as in all other cases when authentication is enabled.